src/test/java/bio/terra/service/dataset/ValidateAssetUnitTest.java

```diff
@@ -0,0 +1,175 @@
+  @MockBean private LoadService loadService;
+  @MockBean private ProfileDao profileDao;
+  @MockBean private StorageTableService storageTableService;
+  @MockBean private BigQueryTransactionPdao bigQueryTransactionPdao;
+  @MockBean private BigQueryDatasetPdao bigQueryDatasetPdao;
+  @MockBean private MetadataDataAccessUtils metadataDataAccessUtils;
+  @MockBean private ResourceService resourceService;
+  @MockBean private GcsPdao gcsPdao;
+  @MockBean private ObjectMapper objectMapper;
+  @MockBean private AzureBlobStorePdao azureBlobStorePdao;
+  @MockBean private ProfileService profileService;
+  @MockBean private UserLoggingMetrics loggingMetrics;
+
+  private String tableName = "tableName1";
+  private String tableName2 = "tableName2";
+  private String col1Name = "col1";
+  private String col2Name = "col2";
+  private String col3Name = "col3";
+  private String relationshipName = "rel1";
+  private Dataset dataset;
+  private AssetModel assetModel;
+
+  @Before
+  public void setup() {
+    dataset =
+        new Dataset()
+            .tables(
+                Arrays.asList(
+                    DatasetFixtures.generateDatasetTable(
+                        tableName, TableDataType.STRING, List.of(col1Name, col2Name)),
+                    DatasetFixtures.generateDatasetTable(
+                        tableName2, TableDataType.STRING, List.of(col3Name))))
+            .relationships(Arrays.asList(new Relationship().name(relationshipName)));
+
+    assetModel =
+        new AssetModel()
+            .name("assetName")
+            .rootTable(tableName)
+            .rootColumn(col1Name)
+            .tables(
+                Arrays.asList(
+                    DatasetFixtures.generateAssetTable(tableName, List.of(col1Name, col2Name))))
+            .follow(Arrays.asList(relationshipName));
+  }
+
+  @Test
+  public void validateAssetModel() {
+    datasetService.validateDatasetAssetSpecification(dataset, assetModel);
+  }
+
+  @Test
+  public void testTwoTables() {
+    assetModel.tables(
+        Arrays.asList(
+            DatasetFixtures.generateAssetTable(tableName, List.of(col1Name, col2Name)),
+            DatasetFixtures.generateAssetTable(tableName2, List.of(col3Name))));
+    datasetService.validateDatasetAssetSpecification(dataset, assetModel);
+  }
+
+  @Test(expected = InvalidAssetException.class)
+  public void testTwoTablesInvalidOverlap() {
+    // col3 is only in table2, not table2
+    assetModel.tables(
+        Arrays.asList(
+            DatasetFixtures.generateAssetTable(tableName, List.of(col1Name, col3Name)),
+            DatasetFixtures.generateAssetTable(tableName2, List.of(col3Name))));
+    testAssetModel(
+        "invalid column", "Column " + col3Name + " does not exist in table " + tableName);
+  }
+
+  @Test(expected = InvalidAssetException.class)
+  public void testInvalidColumn() {
+    String invalidColumn = "InvalidCol";
+    assetModel.tables(
+        Arrays.asList(
```

Comment:
I think we can replace `Arrays.asList()` everywhere with `List.of()`. In new code I think `List.of()` should be preferred over this form, and the `Collections` forms too.
```suggestion
        List.of(
```

Reply:
Ok - I think I've changed all of the instances, but let me know if I've forgotten any.